pkg/virtctl/ssh/native.go

```diff
@@ -0,0 +1,169 @@
+package ssh
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+
+	"github.com/golang/glog"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+	"golang.org/x/term"
+)
+
+func (o *SSH) prepareSSHClient(kind, namespace, name string) (*ssh.Client, error) {
+	streamer, err := o.prepareSSHTunnel(kind, namespace, name)
+	if err != nil {
+		return nil, err
+	}
+
+	conn := streamer.AsConn()
+	addr := fmt.Sprintf("%s/%s.%s", kind, name, namespace)
+	authMethods, err := o.getAuthMethods(kind, namespace, name)
+	if err != nil {
+		return nil, err
+	}
+
+	sshConn, chans, reqs, err := ssh.NewClientConn(conn,
+		addr,
+		&ssh.ClientConfig{
+			HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+			Auth:            authMethods,
+			User:            sshUsername,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return ssh.NewClient(sshConn, chans, reqs), nil
+}
+
+func (o *SSH) getAuthMethods(kind, namespace, name string) ([]ssh.AuthMethod, error) {
+	var (
+		methods []ssh.AuthMethod
+		err     error
+	)
+
+	if len(identityFilePath) < 1 {
```

Comment:
ssh just chains them unconditionally. We can probably do the same. Would it make sense to use `filepath.Join(os.UserHomeDir().ssh/id_rsa)` as default? We should just only ask for the password when it is the turn of the key.